stuff/cron/testing/test_plagiarism_detector.py

```diff
@@ -0,0 +1,267 @@
+                finish_time,
+                check_plagiarism,
+                problemset_id,
+                acl_id,
+            ))
+        Contest.contest_id = cur.lastrowid
+    dbconn.conn.commit()
+
+    # add problemset to contest
+    with dbconn.cursor() as cur:
+        cur.execute(
+            '''
+                UPDATE `Problemsets`
+                SET `contest_id` = (
+                    SELECT `contest_id` FROM
+                    `Contests` 
+                    WHERE `alias` = %s
+                )
+                WHERE problemset_id = %s;
+                    ''', (
+                alias,
+                problemset_id,
+            ))
+    dbconn.conn.commit()
+
+    # setting foreign key check = 0 because of dependencies
+    with dbconn.cursor() as cur:
+        cur.execute('''
+                SET foreign_key_checks = 0;
+                TRUNCATE TABLE Plagiarisms;
+                TRUNCATE TABLE Submission_Log;
+                SET foreign_key_checks = 1;
+                ''')
+
+    # replacing the guid name for new tests to run.
+    guid = 1
+    for user in range(1, no_of_users + 1):
+        for problem in range(1, no_of_problems + 1):
+            gguid = f'{guid:032x}'
+            with dbconn.cursor(buffered=True) as cur:
+                cur.execute(
+                    ''' 
+                        DELETE FROM `Submissions`
+                        WHERE `guid` = %s;
+                        ''', (gguid, ))
+            guid += 1
+        dbconn.conn.commit()
+
+    # add submissions to the contest
+    guid = 1
+    for identity in range(1, no_of_users + 1):
+        for problem in range(1, no_of_problems + 1):
+            with dbconn.cursor() as cur:
+                cur.execute(
+                    '''
+                        INSERT INTO `Submissions`
+                        (`identity_id`,
+                            `problem_id`, `problemset_id`, 
+                            `guid`, `language`, `status`, 
+                            `verdict`, `type`
+                        )
+                        VALUES
+                        (%s, %s, %s, %s, %s, %s, %s,
+                        %s)
+                            ''', (
+                        identity,
+                        problem,
+                        problemset_id,
+                        f'{guid:032x}',
+                        language,
+                        status,
+                        verdict,
+                        ttype,
+                    ))
+                GUID.append(f'{guid:032x}')
+                guid += 1
+
+    dbconn.conn.commit()
+    Contest.GUID = GUID
+
+
+def test_get_contest(dbconn: lib.db.Connection) -> None:
+
+    create_contest(dbconn)
+    contest_id = Contest.contest_id
+    GUID = Contest.GUID
+
+    get_contests_detector = cron.plagiarism_detector.get_contests(dbconn)
+
+    contests: List[int] = [
+        contest['contest_id'] for contest in get_contests_detector
```

Comment:
```suggestion
        c['contest_id'] for c in get_contests_detector
```

otherwise this will overwrite the other `contest`.